overlord/hookstate/ctlcmd/service_scope.go

```diff
@@ -0,0 +1,75 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016-2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package ctlcmd
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/client"
+	"github.com/snapcore/snapd/strutil"
+)
+
+// userAndScopeOptions represents shared options between service operations
+// that change the scope of services affected. This should more or less
+// match what's in cmd/snap.
+type userAndScopeOptions struct {
```

Comment:
we have the same code in cmd mostly?  we have client/clientutil that is used already here to possibly share such things

Reply:
Moved the code into clientutil with dedicated tests